src/Message/SNS/Envelope.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Kfz24\QueueBundle\Message\SNS;
+
+use JMS\Serializer\Annotation as Serializer;
+
+/**
+ * Class Envelope
+ * @package Kfz24\QueueBundle\Message\SNS
+ */
+class Envelope
+{
+    /**
+     * @var MessageInterface
+     * @Serializer\MaxDepth(1)
+     */
+    protected $message;
+
+    /**
+     * @var \DateTimeImmutable
+     * @Serializer\Type("DateTimeImmutable")
+     */
+    protected $createdAt;
```

Comment:
I like the idea of the `MessageInterface`, but I think we are missing the `message` field as it is defined in the `Envelop` schema: https://github.com/kfzteile24/json-schema/blob/master/envelop/v0.json

The Envelop contains 4 required fields:
`"required": [
        "version",
        "type",
        "message",
        "created_at"
    ]`
but right now we are implementing only 3: `createdAt`, `version` and `type`.

So taking as example the `ProductNotificationDto` implemented in the Search PR [Search PR](https://github.com/kfzteile24/search-service/blob/713be79f138cae730fb4a375254705b7b5c6ffb4/search/src/Kfz24/Search/ExportBundle/Dto/ProductNotificationDto.php) the result of serializing the envelope would be:
`[
        "version",
        "type",
        "product_id",
        "sales_channel_id"
        "created_at"
    ]`
instead of 
`[
        "version",
        "type",
"message": [
        "product_id",
        "sales_channel_id"
],
        "created_at"
    ]`

At least this is how I am interpreting the cod, I might be wrong :sweat_smile:

Reply:
Please see https://github.com/kfzteile24/queue-bundle/pull/3/files#r347907772